### DISCUSSION TITLE
Access denied from USGS Earth Explorer M2M?

### DISCUSSION THREAD

**Discussion by peterkobylarz** [earthlab-education/Earth-Analytics-2023-01-Intro#588] [Q&A08 Multispectral Data + Loops + Functions + Conditionals] [answered]
@eculler @nquarder I'm working on the For Loop download for Lincoln Park and Humboldt Park. When I submit the request to download the NAIP imagery using the following code, I'm getting an error from m2m.cr.usgs.gov saying error 403 access denied:
```
    naip_downloader = etee.EarthExplorerDownloader(
        dataset="NAIP", 
        label=row_lower_dash, 
        bbox=bbox,
        start='2021-01-01', 
        end='2021-12-31',
        store_credential=False)
    print(naip_downloader.label)
    print(naip_downloader.bbox.urx, naip_downloader.bbox.ury)
    naip_downloader.submit_download_request()
    naip_downloader.download(override=False)
```
Here's a screenshot of the error page I get referred to:
![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143359014/d99811dd-0f3b-43f1-aede-c1cf12e35414)

I checked my account page on USGS Earth Explorer and it says I have access to the M2M interface through 10/30/2024. Any ideas?

**Comment by nquarder** [comment]
@peterkobylarz can you set up a time w/ @eculler to do some live troubleshooting? She's looking at this over my shoulder right now and thinks we might be stumped! :)   She's keeping office hrs today @ 4pm MT and if that doesn't work for you let's try to find another time. No fear we'll help get this figured out. -NQ

**Reply by nquarder** [reply]
@peterkobylarz this is eculler...I'm also wondering if your credential isn't getting entered correctly. I'd recommend changing it to `store_credential=True` so that you don't have to enter it every time through the loop.

**Comment by peterkobylarz** [comment]
I can jump on office hours at 4 today. I'm pretty sure I'm entering my
credentials correctly, I went to the USGS EE website and tried it there and
can log in.

**Reply by peterkobylarz** [reply, marked as answer]
I reset my password for USGS EE and it worked. Very weird because the credential input in Codespaces seemed to accept my old username/password combo before throwing the error.
